### PR TITLE
queues | Daniel To

### PR DESCRIPTION
**Pull request by theDanielTo**
image

### CODE REVIEW

File: data-structures-queues/take-next-smallest.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/* exported takeNextSmallest */
+
+function takeNextSmallest(queue) {
+  if (typeof queue.peek() === 'undefined') return;
+
+  let current = queue.dequeue();
+
+  while (true) {
+    if (typeof queue.peek() !== 'undefined') {
```
Comment (scbowler) [inline data-structures-queues/take-next-smallest.js:9]: Don't do this, passing `true` to a `while` loop is never a good idea. I know people do it but there is always a better solution.

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, just one update needed. Let me know if you have any questions.

File: data-structures-queues/take-next-smallest.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  container.push(lastVal);
+
+  let found = false;
+  while (typeof queue.peek() !== 'undefined' && !found) {
+    const last = container.length - 1;
+    if (container[last] < queue.peek()) {
+      lastVal = container[last];
+      found = true;
+    } else {
+      container.push(queue.dequeue());
```
Comment (scbowler) [inline data-structures-queues/take-next-smallest.js:17]: Why not just put it to the end of the queue here (`enqueue`)?

**Review by scbowler — APPROVED** [review]
Looks good! 👍 

1 FYI comment, no changes required!

File: data-structures-queues/take-next-smallest.js (hunk 2 above)
Comment (theDanielTo) [reply data-structures-queues/take-next-smallest.js:17]: I had it that way in my previous solution, but I couldn't figure out a way to exit the loop if no value was found that fit the problem's criteria. If we assume a strictly increasing sequence, wouldn't enqueue just keep adding values back into the queue?

**Pull request merged**